cas-server-webapp-support/src/main/java/org/jasig/cas/web/support/CasCookieValueManager.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.cas.web.support;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jasig.cas.util.CipherExecutor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * The {@link CasCookieValueManager} is responsible for...
+ *
+ * @author Misagh Moayyed
+ * @since 4.1
+ */
+public final class CasCookieValueManager implements CookieValueManager {
```

Comment:
Just wondering about the extension capabilities of this class? Shouldn't we be able to choose the extra information we want to rely on?